blog-posts/github-billing-dashboard.md

```diff
@@ -0,0 +1,169 @@
+---
+path: "/blog/github-billing-dashboard/"
+date: "2021-07-15"
+title: "GitHub CSV Billing Dashboard"
+featuredImage: images/space.jpg
+author: "Fabian & Felix"
+authorSummary: "Developers at Satellytes"
+attribution:
+creator: upklyak
+source: https://www.freepik.com/free-vector/space-landscape-night_13748451.htm
+---
+
+
+
+> You can try out the dashboard here: [https://satellytes.github.io/github-billing-dashboard](https://satellytes.github.io/github-billing-dashboard)
+
+We use CI/CD pipelines and automatizations in every project we are responsible for. Especially Github Actions is becoming more popular, as it’s simple to use and growing in functionality. Over the last couple of months, we managed to create multiple projects and repositories that are using Github Actions extensively.
+
+Github Actions provides a generous free tier, so costs for smaller projects are often nonexistent. Once you reach the free tier limits, every Action (and other services) needs to be "paid by use" at the end of the month. Github aggregates all costs and shows them in your settings. But Github doesn't provide charts or a dashboard to get into the details of the costs, for example for historical data.
+
+This can be a problem if you have multiple projects, as it’s not obvious which service causes the costs. If you want to figure that out, you can get a CSV from Github, which is hard to read and analyze. We, therefore, decided to create a dashboard to visualize this CSV and show you the costs per repository and month.
+
+![screenshot of the billing settings in a Github account](./images/screenshot-of-gh-billing-settings.png)
+
+This is all the billing information you get from the Github Settings. No historical data, no expense per service.
+
+![screenshot of the Github billing dashboard with some charts](./images/screenshot-of-gh-dashboard.png)
+
+And this is what you get from the Github Billing Dashboard. Costs per service per month with historic data.
+
+This blog post gives you some insights into how we build it.
+
+## Technologies and Libraries
+
+The dashboard runs locally in the user's browser and thus does not send any data to external servers, even the most recently used files are stored only in the browser's local storage. The dashboard was mainly developed with React and Typescript. For the styling of the dashboard, we used styled-components. In addition, various libraries were used.
+
+### papaparse
+
+[papaparse](https://github.com/mholt/PapaParse) is a CSV parser that converts CSV to JSON. Besides strings, papaparse can process CSV files directly, no matter if they are located locally on the computer or have to be accessed via an external URL.
```

Comment:
```suggestion
[papaparse](https://github.com/mholt/PapaParse) is a CSV parser that converts CSV to JSON. Besides strings, papaparse can process CSV files directly, whether they are located locally on the computer or accessed via an external URL.
```

Reply:
Have you changed the content manually? You are ware of the possibility to use "add suggestion to batch" to collect the proposals you want to accept and then merge them before starting your local other changes? I'm asking because it looks like you did it manually and you could save some time by embracing the feature the next time. I'm going to resolve this now because I have seen the changes. Thanks for accepting it.

Reply:
Yes, I did it manually. Thanks for the hint.